MNT: Structure recipe to show correct outputs

matthewfeickert opened the pull request:
* Use `${{ name }}-suite` for the recipe name to not confuse conda-smithy into expecting there to be a non-static output.
* Use rattler-build `package_contents` to ensure package contains expected files.

<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [N/A] Bumped the build number (if the version is unchanged)
* [N/A] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

matthewfeickert committed:
MNT: Structure recipe to show correct outputs

* Use '${{ name }}-suite' for the recipe name to not confuse conda-smithy
  into expecting there to be a non-static output.
* Use rattler-build package_contents to ensure package contains expected files.

matthewfeickert commented on recipe/recipe.yaml:
```diff
@@ -5,7 +5,7 @@ context:
 
 recipe:
-  name: ${{ name }}
+  name: ${{ name }}-suite
```
For reasons unclear to me, having this name is causing it to show up in the README output, when it should just be the names of the outputs.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

matthewfeickert commented:
This needs https://github.com/conda-forge/conda-smithy/issues/2397 to be addressed first.

matthewfeickert committed:
MNT: Re-rendered with conda-smithy 3.53.3 and conda-forge-pinning 2025.12.16.12.55.14

Other tools:
- conda-build 25.11.1
- rattler-build 0.54.0
- rattler-build-conda-compat 1.4.10
